x-pack/legacy/plugins/ml/server/models/data_visualizer/data_visualizer.ts

```diff
@@ -4,52 +4,142 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
+import { RequestHandlerContext } from 'kibana/server';
 import _ from 'lodash';
+import { ML_JOB_FIELD_TYPES } from '../../../common/constants/field_types';
+import { getSafeAggregationName } from '../../../common/util/job_utils';
 import {
   buildBaseFilterCriteria,
   buildSamplerAggregation,
   getSamplerAggregationsResponsePath,
 } from '../../lib/query_utils';
-import { ML_JOB_FIELD_TYPES } from '../../../common/constants/field_types';
-import { getSafeAggregationName } from '../../../common/util/job_utils';
 
 const SAMPLER_TOP_TERMS_THRESHOLD = 100000;
 const SAMPLER_TOP_TERMS_SHARD_SIZE = 5000;
 const AGGREGATABLE_EXISTS_REQUEST_BATCH_SIZE = 200;
 const FIELDS_REQUEST_BATCH_SIZE = 10;
 
+interface FieldData {
+  fieldName: string;
+  existsInDocs: boolean;
+  stats?: {
+    sampleCount?: number;
+    count?: number;
+    cardinality?: number;
+  };
+}
+
+export interface Field {
+  fieldName: string;
+  type: string;
+  cardinality: number;
+}
+
+interface Distribution {
+  percentiles: any[];
+  minPercentile: number;
+  maxPercentile: number;
+}
+
+interface Aggs {
+  [key: string]: any;
+}
+
+interface Bucket {
+  doc_count: number;
+}
+
+interface NumericFieldStats {
+  fieldName: string;
+  count: number;
+  min: number;
+  max: number;
+  avg: number;
+  isTopValuesSampled: boolean;
+  topValues: Bucket[];
+  topValuesSampleSize: number;
+  topValuesSamplerShardSize: number;
+  median?: number;
+  distribution?: Distribution;
+}
+
+interface StringFieldStats {
+  fieldName: string;
+  isTopValuesSampled: boolean;
+  topValues: Bucket[];
+  topValuesSampleSize: number;
+  topValuesSamplerShardSize: number;
+}
+
+interface DateFieldStats {
+  fieldName: string;
+  count: number;
+  earliest: number;
+  latest: number;
+}
+
+interface BooleanFieldStats {
+  fieldName: string;
+  count: number;
+  trueCount: number;
+  falseCount: number;
+  [key: string]: number | string;
```

Comment:
Is this needed? Don't we just use `falseCount` and `trueCount` ?

Reply:
yeah it's required because of https://github.com/elastic/kibana/pull/56739/files#diff-45e703fa74c02467ebd253aa40c78e0bR876